core/tee/tee_svc_storage.c

```diff
@@ -455,22 +456,36 @@ TEE_Result tee_svc_storage_obj_open(uint32_t storage_id, void *object_id,
 
 	res = tee_svc_storage_read_head(sess, o);
 	if (res != TEE_SUCCESS) {
+		if (res == TEE_ERROR_CORRUPT_OBJECT) {
```

Comment:
tee_svc_storage_read_head result is not only TEE_ERROR_CORRUPT_OBJECT. I delete the corrupt object and its associated handles only if res=TEE_ERROR_CORRUPT_OBJECT.